xonsh/completers/tools.py

```diff
@@ -0,0 +1,33 @@
+import builtins
+
+
+def get_filter_function():
+    csc = builtins.__xonsh_env__.get('CASE_SENSITIVE_COMPLETIONS')
+    if csc:
+        def filt(s, x): return s.startswith(x)
+    else:
+        def filt(s, x): return s.lower().startswith(x.lower())
+    return filt
+
+
+def is_iterable(x):
+    try:
+        _ = iter(x)
+        return True
+    except:
```

Comment:
This should probably yield a type error.  Also,  I think the prefered version of this function is simply `isinstance(x, collections.abs.Iterable)`